lib/internal/modules/esm/worker.js

```diff
@@ -97,6 +98,19 @@ async function customizedModuleWorker(lock, syncCommPort, errorHandler) {
     // so it can detect the exit event.
     const { exit } = process;
     process.exit = function(code) {
+      if (hooks) {
+        for (let i = 0; i < allThreadRegisteredHandlerPorts.length; i++) {
+          const registeredPort = allThreadRegisteredHandlerPorts[i];
+          registeredPort.postMessage(wrapMessage('exit', code ?? process.exitCode));
+        }
+
+        for (const { port, lock: operationLock } of unsettledResponsePorts) {
+          port.postMessage(wrapMessage('exit', code ?? process.exitCode));
```

Comment:
nit: we should cache the result of `wrapMessage('exit', code ?? process.exitCode)` to reuse it in both loops (and outside the loops below)

Reply:
pushed a fix for this one too